pkg/sdk/connection_direct.go

```diff
@@ -30,11 +30,14 @@ var _ Connection = (*directConnection)(nil)
 // or intermediate systems. This is mostly used for test scenarios.
 type directConnection struct {
 	endpoint string
+
+	// insecureAllowCredentialWithHTTP allows sending credentials over HTTP. This is disabled by default.
+	insecureAllowCredentialWithHTTP bool
 }
 
 // NewDirectConnection parses the given endpoint string and returns a direct connection if the endpoint uses the http or
 // https scheme, otherwise it returns an error.
-func NewDirectConnection(endpoint string) (Connection, error) {
+func NewDirectConnection(endpoint string, insecureAllowCredentialWithHTTP bool) (Connection, error) {
```

Comment:
Do we ever set this to true? I'd really not like for this to be part of the API if we don't have a clear usecase for it.

Reply:
We set this to true for some tests: https://github.com/radius-project/radius/blob/main/pkg/portableresources/processors/resourceclient_test.go#L295.